internal/services/storagemover/storage_mover_agent_resource_test.go

```diff
@@ -0,0 +1,180 @@
+	"github.com/hashicorp/go-azure-sdk/resource-manager/storagemover/2023-03-01/agents"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type StorageMoverAgentTestResource struct{}
+
+func TestAccStorageMoverAgent_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_storage_mover_agent", "test")
+	r := StorageMoverAgentTestResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccStorageMoverAgent_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_storage_mover_agent", "test")
+	r := StorageMoverAgentTestResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccStorageMoverAgent_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_storage_mover_agent", "test")
+	r := StorageMoverAgentTestResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccStorageMoverAgent_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_storage_mover_agent", "test")
+	r := StorageMoverAgentTestResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r StorageMoverAgentTestResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := agents.ParseAgentID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	client := clients.StorageMover.AgentsClient
+	resp, err := client.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (r StorageMoverAgentTestResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
```

Comment:
the `provider` block should be in each test, not the template - to allow for overriding it

Reply:
Code has been updated.